eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -1819,6 +1820,15 @@ protected function internalPublishVersion(APIVersionInfo $versionInfo, $publicat
 
         $this->publishUrlAliasesForContent($content);
 
+        if ($this->settings['remove_archived_versions_on_publish']) {
+            $this->deleteOverflownArchivedVersions($contentId);
+        }
+
+        return $content;
+    }
+
+    protected function deleteOverflownArchivedVersions(int $contentId): void
```

Comment:
Naming kinda didn't work for me. After discussing this briefly with @DominikaK we would suggest something more straightforward: 
```suggestion
    protected function deleteArchivedVersionsOverLimit(int $contentId): void
```

Reply:
I was considering the same thing, but opted not to discuss it.